mutex_map.go

```diff
@@ -43,6 +43,23 @@ func (m *MutexMap[K, V]) Delete(key K) {
 	delete(m.real, key)
 }
 
+// DeleteIf deletes every element if the callback returns true.
+// Returns the amount of elements deleted.
+func (m *MutexMap[K, V]) DeleteIf(callback func(key K, value V) bool) int {
+	m.Lock()
+	defer m.Unlock()
+
+	amount := 0
+	for key, value := range m.real {
+		if callback(key, value) {
+			delete(m.real, key)
+			amount++
+		}
+	}
+
+	return amount
+}
```

Comment:
`callback` usually refers to a function to run after an operation completes. In cases like these, where the function is used as a comparison in a set of data, this would usually be called a `predicate`.

```suggestion
func (m *MutexMap[K, V]) DeleteIf(predicate func(key K, value V) bool) int {
	m.Lock()
	defer m.Unlock()

	amount := 0
	for key, value := range m.real {
		if predicate(key, value) {
			delete(m.real, key)
			amount++
		}
	}

	return amount
}
```